### ISSUE TITLE
AutoComplete: Accessibility issues with `<p:column>` custom rendering

### ISSUE THREAD

**Issue by pfumi** [primefaces/primefaces#13854]
### Describe the bug

Issue 2&3 from #13849

![Image](https://github.com/user-attachments/assets/5c85364f-69bc-4f63-a496-f5b9c4964be5)
![Image](https://github.com/user-attachments/assets/509d9e6b-f255-4c4f-b691-0c2027deed67)

### Reproducer

1. Run [primefaces-test.zip](https://github.com/user-attachments/files/20444283/primefaces-test.zip)
2. Click into the field "autocomplete tabular" and type "val1"
3. Problems described in "Describe the bug" arise

### Expected behavior

The problem is the `<tbody>` tag. Elements with `role="option"` have to be direct children of an element with `role="listbox"`.

### PrimeFaces edition

Community

### PrimeFaces version

15.0.3

### JSF implementation

All

### JSF version

4.0

### Java version

11

**Comment by melloware** [comment]
Yeah this stuff was written with TR/TD rendering for custom in 2010 before most ARIA rules existed 😄

**Comment by melloware** [comment]
Oh nice and same for AutoComplete as SelectOne?

**Comment by pfumi** [comment]
AutoComplete has the same structure, so yes

**Comment by melloware** [comment]
OK PR submitted.

**Comment by pfumi** [comment]
**SelectCheckboxMenu:** It has a different structure, both unordered list and tabular.
There's a hidden input field with all the aria attributes and `role="option"`, but it's not a child of the element with `role="listbox"` or `role="group"`, so it's invalid.

**CascaseSelect:** Can't check for tabular on the showcase, but it has multiple accessiblity issues also on the normal view.
For example, the options have no roles, and there are no aria attributes applied.
I am afraid that the structure is quite difficult to solve regarding accessibility.

The issues on all other components (AutoComplete, SelectManyMenu, SelectOneListbox, SelectOneMenu) are solved.

If we want to do something for SelectCheckboxMenu and CascaseSelect, I suggest to create new issues.

**Comment by melloware** [comment]
@pfumi first thank you so much for reporting and testing!

As for **SelectCheckBoxMenu** i worked with @Rene-Stoeckel  in ticket https://github.com/primefaces/primefaces/issues/11850 and he verified with actual screenreaders that all is working well. So I am hesitant to change anything reported by AXE.

**CascaseSelect**: I am all ears on how to fix it if you have recommendations.

**Comment by pfumi** [comment]
Maybe AXE is a bit to strict that option has to be a direct child of listbox.

https://www.w3.org/TR/wai-aria/#option
> Options not associated with a [listbox](https://www.w3.org/TR/wai-aria/#listbox) might not be correctly mapped to an [accessibility API](https://www.w3.org/TR/wai-aria/#dfn-accessibility-api).

The W3C specification says it "might" be a problem, so if screenreaders interpret the option correctly, then I would leave it like that.
